org.eclipse.scout.rt.server.commons/src/main/java/org/eclipse/scout/rt/server/commons/GlobalTrustManager.java

```diff
@@ -77,16 +84,44 @@ public void installGlobalTrustManager(String protocol, String tmAlgorithm) {
     X509TrustManager globalTrustManager;
     try {
       globalTrustManager = createGlobalTrustManager(tmAlgorithm, getAllTrustedCertificates());
-
       SSLContext sslContext = SSLContext.getInstance(protocol);
-      sslContext.init(null, new TrustManager[]{globalTrustManager}, SecurityUtility.createSecureRandom());
+      sslContext.init(loadKeyManagers(), new TrustManager[]{globalTrustManager}, SecurityUtility.createSecureRandom());
       SSLContext.setDefault(sslContext);
     }
     catch (Exception e) {
       throw new ProcessingException("could not install global trust manager.", e);
     }
   }
 
+  /**
+   * Loads a list of {@link KeyManager} based on {@code javax.net.ssl.*} system properties.
+   *
+   * @see sun.security.ssl.SSLContextImpl.DefaultManagersHolder#getKeyManagers()
+   */
+  @SuppressWarnings("squid:S1168")
+  protected KeyManager[] loadKeyManagers() throws KeyStoreException, IOException, NoSuchAlgorithmException, CertificateException, UnrecoverableKeyException, NoSuchProviderException {
+    String keystoreLocation = System.getProperty("javax.net.ssl.keyStore");
```

Comment:
Maybe use the ConfigUtility instead of System.getProperty? This would allow us specifying this value using config.properties or environment variables (which might be handy for containers)? What do you think?

Reply:
Added